pkg/frontend/compiler_context.go

```diff
@@ -268,40 +269,48 @@ func (tcc *TxnCompilerContext) Resolve(dbName string, tableName string) (*plan2.
 	return tcc.getTableDef(ctx, table, dbName, tableName, sub)
 }
 
-func (tcc *TxnCompilerContext) ResolveUdf(name string, args []*plan.Expr) (string, error) {
+func (tcc *TxnCompilerContext) ResolveUdf(name string, args []*plan.Expr) (body string, err error) {
 	var expectInvalidArgErr bool
 	var expectedInvalidArgLengthErr bool
 	var badValue string
 	var argstr string
-	var body string
 	var sql string
-	var err error
 	var erArray []ExecResult
 
 	ses := tcc.GetSession()
 	ctx := ses.GetRequestContext()
+
+	err = inputNameIsInvalid(ctx, name)
+	if err != nil {
+		return "", err
+	}
+
 	bh := ses.GetBackgroundExec(ctx)
 	defer bh.Close()
 
 	err = bh.Exec(ctx, "begin;")
+	defer func() {
+		err = finishTxn(ctx, bh, err)
```

Comment:
here the err as params to finishTxn, is the err from bh.Exec(ctx, "begin;").   I think you need the last err before return.    Is this right?

If you think I am wrong, please find some go expert to review this and confirm.

Reply:
the `err` is defined at the `func (tcc *TxnCompilerContext) ResolveUdf(name string, args []*plan.Expr) (body string, err error)`. it is shared among the function.  it is returned the last err always.